### PR TITLE
LibWeb: Implement HTML{Image,Script,Link}.fetchPriority

### PR DESCRIPTION
**Pull request by jamierocks**

### CODE REVIEW

File: Userland/Libraries/LibWeb/HTML/HTMLLinkElement.cpp [hunk 1]
```diff
@@ -234,33 +236,41 @@ HTMLLinkElement::LinkProcessingOptions HTMLLinkElement::create_link_options()
 JS::GCPtr<Fetch::Infrastructure::Request> HTMLLinkElement::create_link_request(HTMLLinkElement::LinkProcessingOptions const& options)
 {
     // 1. Assert: options's href is not the empty string.
+    VERIFY(options.href != "");
 
-    // FIXME: 2. If options's destination is not a destination, then return null.
+    // 2. If options's destination is null, then return null.
+    if (!options.destination.has_value())
+        return nullptr;
```
Comment (tcl3) [inline Userland/Libraries/LibWeb/HTML/HTMLLinkElement.cpp:243]: I believe that this is the change that is causing the tests to fail. This should either be removed for now, or the correct `options.destination` value should be set `HTMLLinkElement::create_link_options()`.

File: Userland/Libraries/LibWeb/HTML/HTMLLinkElement.cpp [hunk 2]
```diff
@@ -234,33 +236,41 @@ HTMLLinkElement::LinkProcessingOptions HTMLLinkElement::create_link_options()
 JS::GCPtr<Fetch::Infrastructure::Request> HTMLLinkElement::create_link_request(HTMLLinkElement::LinkProcessingOptions const& options)
 {
     // 1. Assert: options's href is not the empty string.
+    VERIFY(options.href != "");
```
Comment (tcl3) [inline Userland/Libraries/LibWeb/HTML/HTMLLinkElement.cpp:239]: We generally use `is_empty()` for these kinds of checks, so `VERIFY(!options.href.is_empty())` would be better here.

**Review by tcl3 — CHANGES_REQUESTED** [review]
This change is causing some `HTMLLinkElement`  related tests to fail.

See the CI output [here](https://github.com/LadybirdWebBrowser/ladybird/actions/runs/9355458834/job/25756994577?pr=24)

You should be able to run the LibWeb tests locally by this command from the root directory: `./Meta/ladybird.sh test LibWeb`

File: Userland/Libraries/LibWeb/HTML/HTMLLinkElement.cpp (hunk 1 above)
Comment (jamierocks) [reply Userland/Libraries/LibWeb/HTML/HTMLLinkElement.cpp:243]: I think you're probably right, I'll leave this as a FIXME and handle destinations in another pr.

File: Userland/Libraries/LibWeb/HTML/HTMLLinkElement.cpp (hunk 2 above)
Comment (jamierocks) [reply Userland/Libraries/LibWeb/HTML/HTMLLinkElement.cpp:239]: I'll make this change :)

**Comment by tcl3** [comment]
We have an existing `crossOrigin-attribute.html` test for the `crossOrigin` attribute. It would be great if you could add `link` and `img` to that test. It is kind of repetitive, but it's a good way to catch future regressions if we ever change the way the attribute is implemented.

Other than that, LGTM!

**Review by tcl3 — APPROVED** [review]

**Pull request merged**